Makefile.dep

```diff
@@ -1057,44 +1041,3 @@ ifneq (,$(filter test_utils_interactive_sync,$(USEMODULE)))
     DEFAULT_MODULE += test_utils_interactive_sync_shell
   endif
 endif
-
-# recursively catch transitive dependencies
-USEMODULE := $(sort $(USEMODULE))
-USEPKG := $(sort $(USEPKG))
-ifneq ($(OLD_USEMODULE) $(OLD_USEPKG),$(USEMODULE) $(USEPKG))
-  include $(RIOTBASE)/Makefile.dep
-else
-
-  # set all USED periph_% init modules as DEFAULT_MODULE
-  ifneq (,$(filter periph_init, $(USEMODULE)))
-      # To enable adding periph_% modules through the environment we cant use
-      # FEATURES_USED since the MODULE might be added directly as USEMODULE
-      PERIPH_MODULES_NO_INIT = periph_init% periph_common
-      PERIPH_MODULES = $(filter periph_%,$(USEMODULE))
-      PERIPH_INIT_MODULES = $(subst periph_,periph_init_,\
-        $(filter-out $(PERIPH_MODULES_NO_INIT),$(PERIPH_MODULES)))
-      DEFAULT_MODULE += $(PERIPH_INIT_MODULES)
-  endif
-
-  # add periph_init_% modules to USEMODULE unless disabled
-  ifneq (,$(filter periph_init, $(USEMODULE)))
-    USEMODULE += $(filter $(PERIPH_INIT_MODULES),\
-      $(filter-out $(DISABLE_MODULE),$(DEFAULT_MODULE)))
-  endif
-
-  # Add auto_init_% DEFAULT_MODULES. This is done after the recursive cach since
-  # none of these modules can trigger dependency resolution.
-  ifneq (,$(filter auto_init,$(USEMODULE)))
-    USEMODULE += $(filter auto_init_%,$(filter-out $(DISABLE_MODULE),$(DEFAULT_MODULE)))
-  endif
-
-  # Add test_utils_interactive_sync_shell
-  ifneq (,$(filter test_utils_interactive_sync,$(USEMODULE)))
-    USEMODULE += $(filter test_utils_interactive_sync_%, \
-      $(filter-out $(DISABLE_MODULE),$(DEFAULT_MODULE)))
-  endif
```

Comment:
Note that this global disable mechanism although it didn't work for `auto_init_%` added during the first iteration through `Makefile.dep`, worked for those added in further iteration. This could mean that the output of `save_all_dependencies` could have some extra `auto_init` modules. I fixed that in #13785 with 1ea4fc4.